Sources/BraveUI/SwiftUI/WebImageReader.swift

```diff
@@ -5,82 +5,70 @@
 
 import SwiftUI
 import SDWebImage
-import SDWebImageSVGNativeCoder
 
-private class WalletWebImageManager: ObservableObject {
-  /// loaded image, note when progressive loading, this will published multiple times with different partial image
-  @Published public var image: UIImage?
-  /// loaded image data, may be nil if hit from memory cache. This will only published once loading is finished
-  @Published public var isFinished: Bool = false
-  /// loading error
-  @Published public var error: Error?
-
-  private var manager = SDWebImageManager.shared
-  private var operation: SDWebImageOperation?
+public protocol WebImageDownloaderType: AnyObject {
+  func downloadImage(url: URL) async -> (UIImage?, Error?)
```

Comment:
API design nit: swap to a throwing function instead:

```swift
func downloadImage(url: URL) async throws -> UIImage
```

Either it downloads an image successfully or there's an error. No need to deal with optionals or tuples.

Reply:
`Error` has been removed in [8e4439d](https://github.com/brave/brave-ios/pull/7881/commits/8e4439db703e9ca9718fade5121652e75ebc5e68)